util/httputil/httputil.go

```diff
@@ -75,3 +76,16 @@ func findRemoteAddr(r *http.Request, v iputil.IPValidator) (net.IP, iputil.IPVer
 	}
 	return nil, iputil.IPvUnknown
 }
+
+func ConvertUrlParamsToMap(url *url.URL) map[string]string {
+	rawQuery := url.RawQuery
+	rawQuerySplit := strings.Split(rawQuery, "&")
+	ampData := make(map[string]string, 0)
+	for _, kv := range rawQuerySplit {
+		kvSplit := strings.Split(kv, "=")
+		if len(kvSplit) == 2 {
+			ampData[kvSplit[0]] = kvSplit[1]
+		}
+	}
+	return ampData
```

Comment:
Please do not reimplement query parsing. It's available in [url.Query()](https://pkg.go.dev/net/url#example-URL.Query)

Reply:
Agree! I didn't know about this function, modified!